### PR TITLE
Add scheduler API/controller - no logical implementation

### PR DESCRIPTION
**Pull request by CoderKevinZhang**
**What type of PR is this?**
> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by kimeunju108 — CHANGES_REQUESTED** [review]

**Review by kimeunju108 — CHANGES_REQUESTED** [review]
scheduler function not working

File: globalscheduler/controllers/scheduler/scheduler_controller.go [hunk 1]
```diff
@@ -237,8 +243,35 @@ func (sc *SchedulerController) syncHandler(key string) error {
 	return nil
 }
 
+func (sc *SchedulerController) addScheduler(obj interface{}) {
+	sc.enqueueScheduler(obj)
```
[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply globalscheduler/controllers/scheduler/scheduler_controller.go:247]: Let me do the research on how arktos controllers did it

[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply globalscheduler/controllers/scheduler/scheduler_controller.go]: I have already added event type in my controller, thanks.

**Review by kimeunju108 — APPROVED** [review]
LGTM

**Review by jshaofuturewei — APPROVED** [review]
Can you squash your commits?  There are too many of them

**Pull request merged**